tasks/check-interface.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: Check network interface has a network name
+  fail:
+    msg: >
+      The interface definition {{ interface }} has type 'network', but does not have
+      a network name defined.
+  when:
+    - interface.type is defined
+    - interface.type == 'network'
```

Comment:
interface.type is optional, so how about:

`interface.type is not defined or interface.type == 'network'`

Reply:
Ah yeah, good point.